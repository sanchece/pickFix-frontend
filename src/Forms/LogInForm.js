import React, { useState } from "react";
import { useHistory } from "react-router-dom";
import {
  Card,
  Container,
  Row,
  Col,
  Button,
  Stack,
  InputGroup,
  FormControl,
  Dropdown,
  DropdownButton,
  Form,
} from "react-bootstrap/";
import "bootstrap/dist/css/bootstrap.min.css";
const LogInForm = ({ logIn }) => {
  const history = useHistory();

  const [logInData, setLogInData] = useState({
    email: "",
    password: "",
    userType: "customers",
  });
  async function handleSubmit(e) {
    e.preventDefault();
    let res = await logIn(logInData);
    if (res.success) {
      history.push("/profile");
    } else {
      alert("Incorrect Email/Password/UserType");
    }
  }
  async function handleChange(e) {
    const { name, value } = e.target;
    console.log(logInData);
    setLogInData((data) => ({ ...data, [name]: value }));
  }

  function capitalizeFirstLetter(string) {
    return string.charAt(0).toUpperCase() + string.slice(1);
  }
  return (
    <Container style={{ margin: "100px 0px 0px 0px" }} fluid>
      <Row>
        <Col lg={4}></Col>
        <Col lg={4}>
          <Card
            className="d-flex align-items-center justify-content-center"
            border=""
            // style={{ padding: "10px 0px 30px 0px", width: "25em" }}
          >
            <Card.Body>
              <Card.Title className="d-flex mb-3 justify-content-center">
                Log In
              </Card.Title>
                <Form onSubmit={handleSubmit}>
                  {Object.keys(logInData).map((key, i) => {
                    if (key !== "userType" && key !=="password") {
                      return (
                        <div key={`${i}`}>
                          <InputGroup className="mb-3">
                            <InputGroup.Text id="basic-addon1">
                              {capitalizeFirstLetter(key)}
                            </InputGroup.Text>
                            <FormControl
                              placeholder={`Enter ${capitalizeFirstLetter(
                                key
                              )} Here`}
                              name={`${key}`}
                              value={logInData[key]}
                              onChange={handleChange}
                            />
                          </InputGroup>
                        </div>
                      );
                    } 
                    else if (key === "password") {
                      return (
                        <div key={`${i}`}>
                        <InputGroup className="mb-3">
                          <InputGroup.Text id="basic-addon1">
                            {capitalizeFirstLetter(key)}
                          </InputGroup.Text>
                          <FormControl
                            placeholder={`Enter ${capitalizeFirstLetter(
                              key
                            )} Here`}
                            name={`${key}`}
                            value={logInData[key]}
                            onChange={handleChange}
                            type="password"
                          />
                        </InputGroup>
                      </div>
                      );}
                    else if (key === "userType") {
                      return (
                        <div key={`${i}`}>
                          <InputGroup className="mb-3">
                            <InputGroup.Text id="basic-addon1">
                              {capitalizeFirstLetter(key)}
                            </InputGroup.Text>
                            <Form.Select
                              name="userType"
                              id="userType"
                              onChange={handleChange}
                            >
                              <option value="customers">Customer</option>
                              <option value="contractors">Contractor</option>
                            </Form.Select>
                          </InputGroup>
                        </div>
                      );
                    }
                    return <div></div>;
                  })}
                  <div className="d-grid gap-2">
                    <Button onClick={handleSubmit}>Log In</Button>
                  </div>
                </Form>
            </Card.Body>
          </Card>
        </Col>
        <Col lg={4}></Col>
      </Row>
    </Container>
  );
};

export default LogInForm;
