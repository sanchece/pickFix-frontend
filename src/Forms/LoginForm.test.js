import React from "react";
import { render } from "@testing-library/react";
import LogInForm from "./LoginForm";
import { MemoryRouter } from "react-router";

it("matches snapshot", function () {
  const { asFragment } = render(
      <MemoryRouter>
        <LogInForm />
      </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});
