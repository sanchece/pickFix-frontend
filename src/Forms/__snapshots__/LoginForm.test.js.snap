// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container-fluid"
    style="margin: 100px 0px 0px 0px;"
  >
    <div
      class="row"
    >
      <div
        class="col-lg-4"
      />
      <div
        class="col-lg-4"
      >
        <div
          class="d-flex align-items-center justify-content-center card"
        >
          <div
            class="card-body"
          >
            <div
              class="d-flex mb-3 justify-content-center card-title h5"
            >
              Log In
            </div>
            <form
              class=""
            >
              <div>
                <div
                  class="mb-3 input-group"
                >
                  <span
                    class="input-group-text"
                    id="basic-addon1"
                  >
                    Email
                  </span>
                  <input
                    class="form-control"
                    name="email"
                    placeholder="Enter Email Here"
                    value=""
                  />
                </div>
              </div>
              <div>
                <div
                  class="mb-3 input-group"
                >
                  <span
                    class="input-group-text"
                    id="basic-addon1"
                  >
                    Password
                  </span>
                  <input
                    class="form-control"
                    name="password"
                    placeholder="Enter Password Here"
                    type="password"
                    value=""
                  />
                </div>
              </div>
              <div>
                <div
                  class="mb-3 input-group"
                >
                  <span
                    class="input-group-text"
                    id="basic-addon1"
                  >
                    UserType
                  </span>
                  <select
                    class="form-select"
                    id="userType"
                    name="userType"
                  >
                    <option
                      value="customers"
                    >
                      Customer
                    </option>
                    <option
                      value="contractors"
                    >
                      Contractor
                    </option>
                  </select>
                </div>
              </div>
              <div
                class="d-grid gap-2"
              >
                <button
                  class="btn btn-primary"
                  type="button"
                >
                  Log In
                </button>
              </div>
            </form>
          </div>
        </div>
      </div>
      <div
        class="col-lg-4"
      />
    </div>
  </div>
</DocumentFragment>
`;
